app/services/form1010cg/service.rb

```diff
@@ -3,25 +3,85 @@
 # This service manages the interactions between CaregiversAssistanceClaim, CARMA, and Form1010cg::Submission.
 module Form1010cg
   class Service
-    def submit_claim!(claim_data)
-      claim = SavedClaim::CaregiversAssistanceClaim.new(claim_data)
+    attr_reader :claim
+
+    NOT_FOUND = 'NOT_FOUND'
+
+    def initialize(claim)
       claim.valid? || raise(Common::Exceptions::ValidationErrors, claim)
 
-      carma_submission = CARMA::Models::Submission.from_claim(claim)
+      @claim = claim
+      @cached_icns = {}
+    end
 
-      carma_submission.submit!
+    def process_claim!
+      assert_veteran_status
+
+      carma_submission = CARMA::Models::Submission.from_claim(claim, build_metadata).submit!
 
       Form1010cg::Submission.new(
         carma_case_id: carma_submission.carma_case_id,
         submitted_at: carma_submission.submitted_at
       )
     end
 
+    def assert_veteran_status
+      raise_unprocessable if icn_for('veteran') == NOT_FOUND
+    end
+
+    def build_metadata
+      claim.form_subjects.each_with_object({}) do |namespace, acc|
+        icn = icn_for(namespace)
+        acc[namespace.to_sym] = {
+          icn: icn == 'NOT_FOUND' ? nil : icn
```

Comment:
Maybe reuse your `NOT_FOUND` constant? Or just use a string here and remove the constant everywhere else